File: ci-cd pallavi/backend/app/routes/health_routes.py
from flask import Blueprint, jsonify
from app.config.db import query_db
import datetime

health_bp = Blueprint('health', __name__)

@health_bp.route('/', methods=['GET'])
def health_check():
    status = {
        "status": "UP",
        "timestamp": datetime.datetime.now().isoformat(),
        "services": {
            "api": "OK",
            "database": "UNKNOWN"
        }
    }
    
    # Check DB
    db_test = query_db("SELECT 1", one=True)
    if db_test:
        status["services"]["database"] = "OK"
    else:
        status["status"] = "DEGRADED"
        status["services"]["database"] = "DOWN"
        
    return jsonify(status), 200 if status["status"] == "UP" else 503

@health_bp.route('/test', methods=['GET'])
def test_api():
    return jsonify({"message": "Flask API is working correctly"}), 200

File: ci-cd pallavi/backend/routes/healthRoutes.js
const express = require('express');
const router = express.Router();
const db = require('../config/db');

// @route   GET /api/health
// @desc    Check API and Database status
// @access  Public
router.get('/', async (req, res) => {
  const healthStatus = {
    status: 'UP',
    timestamp: new Date().toISOString(),
    services: {
      api: 'OK',
      database: 'UNKNOWN'
    }
  };

  try {
    // Attempt a simple query to verify DB connectivity
    await db.query('SELECT 1');
    healthStatus.services.database = 'OK';
    res.status(200).json(healthStatus);
  } catch (error) {
    healthStatus.status = 'DEGRADED';
    healthStatus.services.database = 'DOWN';
    healthStatus.error = error.message;
    res.status(503).json(healthStatus);
  }
});

// @route   GET /api/health/test
// @desc    Simple test endpoint to verify API is reachable
router.get('/test', (req, res) => {
  res.json({ message: 'API is working correctly' });
});

module.exports = router;
